From 226181ae43f48750d0a15e088b7cee6f1825d35d Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 7 Apr 2024 21:44:15 +0800
Subject: [PATCH 45/45] [LOONGARCH64] fix(amdgpu_drv.c): disable dpm by default

AMD Radeon GCN 1-4th generation GPUs do not function reliably with DPM
(dynamic power management) enabled, causing frequent resets under load or when
paired with high-resolution/high-refresh-rate monitors. This is probably
caused by a design flaw in Loongson's 7A-series chipset, where DMA data
preceded interrupt requests.

Disable this on LoongArch by default to prevent headaches until otherwise
fixed upstream.
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
index e8fe65cee40b..60c5b215d35e 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
@@ -142,7 +142,7 @@ int amdgpu_hw_i2c;
 int amdgpu_pcie_gen2 = -1;
 int amdgpu_msi = -1;
 char amdgpu_lockup_timeout[AMDGPU_MAX_TIMEOUT_PARAM_LENGTH];
-int amdgpu_dpm = -1;
+int amdgpu_dpm = 0;
 int amdgpu_fw_load_type = -1;
 int amdgpu_aspm = -1;
 int amdgpu_runtime_pm = -1;
@@ -338,8 +338,8 @@ module_param_string(lockup_timeout, amdgpu_lockup_timeout, sizeof(amdgpu_lockup_
 /**
  * DOC: dpm (int)
  * Override for dynamic power management setting
- * (0 = disable, 1 = enable)
- * The default is -1 (auto).
+ * (0 = disable, 1 = enable, -1 = auto)
+ * The default is 0 (disable).
  */
 MODULE_PARM_DESC(dpm, "DPM support (1 = enable, 0 = disable, -1 = auto)");
 module_param_named(dpm, amdgpu_dpm, int, 0444);
-- 
2.44.0

